class6.1/main.cpp

```diff
@@ -0,0 +1,187 @@
+        if (orangeThread_.joinable()) orangeThread_.join();
+    }
+
+private:
+    void generateGreenMessages() {
+        while (true) {
+            GreenMessage msg = generateGreenMessage();
+            dispatcher_.publish(msg);
+            std::this_thread::sleep_for(std::chrono::milliseconds(376));
+        }
+    }
+
+    void generateBlueMessages() {
+        while (true) {
+            BlueMessage msg = generateBlueMessage();
+            dispatcher_.publish(msg);
+            std::this_thread::sleep_for(std::chrono::milliseconds(473));
+        }
+    }
+
+    void generateOrangeMessages() {
+        while (true) {
+            OrangeMessage msg = generateOrangeMessage();
+            dispatcher_.publish(msg);
+            std::this_thread::sleep_for(std::chrono::milliseconds(756));
+        }
+    }
+
+    GreenMessage generateGreenMessage() {
+        static std::random_device rd;
+        static std::mt19937 gen(rd());
+        static std::uniform_int_distribution<> counterDist(1, 100);
+
+        std::string msg = "Green Message";
+        int counter = counterDist(gen);
+        return { msg, counter };
+    }
+
+    BlueMessage generateBlueMessage() {
+        static std::random_device rd;
+        static std::mt19937 gen(rd());
+        static std::uniform_real_distribution<> valueDist(0.0, 1.0);
+
+        double value1 = valueDist(gen);
+        double value2 = valueDist(gen);
+        return { value1, value2 };
+    }
+
+    OrangeMessage generateOrangeMessage() {
+        static std::random_device rd;
+        static std::mt19937 gen(rd());
+        static std::uniform_int_distribution<> numberDist(1, 100);
+        static std::uniform_real_distribution<> valueDist(0.0, 1.0);
+
+        std::string str1 = "Orange";
+        std::string str2 = "Message";
+        int number = numberDist(gen);
+        double value = valueDist(gen);
+        return { str1, str2, number, value };
+    }
+
+    MessageDispatcher& dispatcher_;
+    std::thread greenThread_;
+    std::thread blueThread_;
+    std::thread orangeThread_;
+};
+
+// Example subscriber implementation
+class ExampleSubscriber : public MessageSubscriber {
+public:
+    void handleMessage(const GreenMessage& msg) override {
+        std::cout << "Received GreenMessage: " << msg.message << " " << msg.counter << std::endl;
+    }
+
+    void handleMessage(const BlueMessage& msg) override {
+        std::cout << "Received BlueMessage: " << msg.value1 << " " << msg.value2 << std::endl;
+    }
+
+    void handleMessage(const OrangeMessage& msg) override {
+        std::cout << "Received OrangeMessage: " << msg.str1 << " " << msg.str2 << " " << msg.number << " " << msg.value << std::endl;
+    }
+};
+
+int main() {
+    MessageDispatcher dispatcher;
+    ExampleSubscriber subscriber;
+
+    dispatcher.subscribe(&subscriber);
+
+    MessageGenerator generator(dispatcher);
+    generator.startGenerating();
+
+  
+    std::this_thread::sleep_for(std::chrono::seconds(10));
```

Comment:
Minor: we can name that delay and make it const/constexpr somewhere in this function - it will make your code readable(we will get some info from name what this delay is responsible for) and it will be easier to reuse or change it.